Release notes for v1.2 (#5054)
gammapy/gammapy: docs/release-notes/v1.2.rst

.. _gammapy_1p2_release:

1.2 (February XXth 2024)
------------------------

- Released February XX, 2024
- 24 contributors
- 178 pull requests since v1.1 (not all listed below)
- yz closed issues

Summary
~~~~~~~

New features
~~~~~~~~~~~~

- New helper classes have been added to perform event sampling.
- A checksum option has been added on read and write methods in Gammapy. It follows the

>   FITS standard and reuses the astropy methods and behavior. A checksum for yaml file

Reviewer: ```suggestion
  FITS standard and reuses the astropy methods and behaviour. A checksum for yaml file
```

  has been introduced as well.
- Total observation time map.

*gammapy.catalogs*

- Update 4FGL to DR4.
- Added 1LHAASO catalog.

*gammapy.data*

- A general scheme for metadata support has been introduced. The ``Metadata`` base
  class has been designed according to PIG X.
- Added a function to remove a time interval from a ``GTI``.
- Added function to export part of a ``DataStore`` to an IVAO compliant ObsCore table.

> 

Reviewer: There was the "ObservationsEventsSampler" class

*gammapy.makers*

- Prototype support for asymmetric IRF in Gammapy's ``Maker`` classes.
  A tutorial exposing how to create such IRFs has been added.

> 

Reviewer: The `SafeMaskMaker` at DL3 level

*gammapy.maps*

> - Implement ``TimeMapAxis.pix_to_coord()``

Reviewer 2: ```suggestion
- Implement ``TimeMapAxis.pix_to_coord()``
- Implement ``TimeMapAxis.to_gti()``
```

*gammapy.modeling*

- Added ``Prior`` class.
- Add function to determine pivot energy for all spectral models.

> - Added position as a parameter for the ``TemplateSpatialModel``. Add a caveat.

Reviewer 2: ```suggestion
- Added position as a parameter for the ``TemplateSpatialModel``. CAVEAT: results are correct only when the fitted position is close to the map center.
- Added Spatial parameters in FoVBackgroundModels
- SkyModel evaluation now supports a TimeMapAxis
- Adapt FluxPointsDataset to directly fit lightcurves
```

*gammapy.estimators*

- Add a ``slice_by_coord()`` function on ``FluxMaps``.
- Introduce timing utility functions: point to point flux variance, fractional excess
  variance
- Add optional sensitivity estimation in ``ExcessMapEstimator``.

> 

Reviewer: The `EnergyDependentMorphologyEstimator`

*gammapy.visualization*

- Add a plot function for the distribution of ``Map`` data.

API changes
~~~~~~~~~~~

- Freeze source parameters in ``FluxEstimator`` classes.
- The ``norm`` parameter is now passed as an argument to the various flux estimators.
  ``Parameter.is_norm`` is now deprecated.
- The default index of ``ExpCutoffPowerlawNormSpectralModel`` has been changed to 0
  for consistency with the ``PowerlawNormSpectralModel``.

Bug fixes and improvements
~~~~~~~~~~~~~~~~~~~~~~~~~~

- Correct ``MapDataset.info_dict()`` to use background model rather than IRF background when
  giving excess counts and significance.
- Import ray only when needed.
- Added information on number of degrees of freedom on ``FluxMaps`` and ``FluxPoints`` objects.
- Reduced memory usage of ``MapEvaluator`` and ``PSFMap.get_psf_kernel()``.
- Added support for multiprocessing in ``FluxProfileEstimator``.
- Added multiprocessing for ``WcsNDMap`` convolution.
- Add a context manager for multiprocessing.
- Add a faster reprojection method : ``reproject_by_image``.

> - Use interpolation for dardk matter mass. Add Zhao profile and

Reviewer: ```suggestion
- Use interpolation for dark matter mass. Add Zhao profile and
```

- [#4879] Fix energy dependent temporal model simulation (Quentin Remy)
- [#4854] Notebook to sphinx-gallery script (REGEARD Maxime)
- [#4851] Parallel support for WcsNDMap map convolution (Quentin Remy)
- [#4850] Add utility function to split dataset into multiple datasets (Quentin Remy)
- [#4849] Add TimeMapAxis.to_gti() (Atreyee Sinha)
- [#4847] Variability tutorial (Claudio Galelli)
- [#4845] Add context manager for multiprocessing configuration (Quentin Remy)
- [#4837] Add checksum argument to gammapy products write functions (Régis Terrier)
- [#4835] Management of metadata for `Models` (Bruno Khélifi)
- [#4834] Adding prior stat sum to datasets (Katrin Streil)
- [#4829] Caching gti and radmax (REGEARD Maxime)
- [#4828] Adapt SkyModel to evaluate on TimeMapAxis (Atreyee Sinha)
- [#4822] Add a function to delete a time interval from GTI (Claudio Galelli)
- [#4817] Computation of total observation time map (Atreyee Sinha)
- [#4814] Introduce a function to compute the doubling/halving time for a lightcurve (Claudio Galelli)
- [#4810] Adding a tutorial for observational clustering (Astro-Kirsty)
- [#4808] adding `Observations` in memory generator (REGEARD Maxime)

> - [#4805] Description of the arguments of the class `Observation' (Bruno Khélifi)

Author: ```suggestion
- [#4805] Description of the arguments of the class `Observation` (Bruno Khélifi)
```
